cookbooks/aws-parallelcluster-slurm/resources/munge_key_manager.rb

```diff
@@ -70,3 +70,58 @@
     end
   end
 end
+
+action :update_munge_key do
+  bash 'remove_current_munge_key' do
+    user 'root'
+    group 'root'
+    cwd '/tmp'
+    code <<-REMOVE_CURRENT_MUNGE_KEY
+      if [ -f "/etc/munge/munge.key" ]; then
+        rm -f /etc/munge/munge.key
+      fi
+    REMOVE_CURRENT_MUNGE_KEY
+  end
+
+  if new_resource.munge_key_secret_arn
+    bash 'fetch_and_decode_munge_key' do
+      user 'root'
+      group 'root'
+      cwd '/tmp'
+      code <<-FETCH_AND_DECODE
+        # Get encoded munge key from secrets manager
+        encoded_key=$(aws secretsmanager get-secret-value --secret-id #{new_resource.munge_key_secret_arn} --query 'SecretString' --output text --region #{node['cluster']['region']})
+        # If encoded_key doesn't have a value, error and exit
+        if [ -z "$encoded_key" ]; then
+          echo "Error fetching munge key from Secrets Manager or the key is empty"
+          exit 1
+        fi
+
+        # Decode munge key and write to /etc/munge/munge.key
+        decoded_key=$(echo $encoded_key | base64 -d)
+        if [ $? -ne 0 ]; then
+          echo "Error decoding the munge key with base64"
+          exit 1
+        fi
+
+        echo "$decoded_key" > /etc/munge/munge.key
+
+        # Set ownership on the key
+        chown #{node['cluster']['munge']['user']}:#{node['cluster']['munge']['group']} /etc/munge/munge.key
+        # Enforce correct permission on the key
+        chmod 0600 /etc/munge/munge.key
+      FETCH_AND_DECODE
+    end
+  else
+    bash 'generate_munge_key' do
+      user node['cluster']['munge']['user']
+      group node['cluster']['munge']['group']
+      cwd '/tmp'
+      code <<-GENERATE_KEY
+        set -e
+        /usr/sbin/mungekey --verbose
+        chmod 0600 /etc/munge/munge.key
+      GENERATE_KEY
+    end
```

Comment:
This block of code is used during the `setup` action as well.
Can we move it to a shared function?

Reply:
Same case for the block of code to fetch and decode the munge key.

Reply:
Done. After removing the checking if the munge key exists section, the update section and the setup section looks the same. But I think we can keep them for now. Because later we may add new logic to update section -> Maintaining a backup copy of the existing Munge key to reuse in case of specific failures